packages/blocks/src/store/reducer.js

```diff
@@ -328,6 +328,9 @@ export const unregisteredFallbackBlockName = createBlockNameSetterReducer(
 export const groupingBlockName = createBlockNameSetterReducer(
 	'SET_GROUPING_BLOCK_NAME'
 );
+export const sectionRootBlockName = createBlockNameSetterReducer(
+	'SET_SECTION_ROOT_BLOCK_NAME'
+);
```

Comment:
What is this about?

Reply:
I don't think we should be introducing APIs for this feature basically.

Reply:
The problem this is trying to solve is how we can know whether we are editing a post/page template or the post/page itself. Because zoom-out is a feature of the generic block editor package it has to work without knowing the implementation details of WordPress, so this is how we are telling it what context we are in. Do you have another idea?

Reply:
Do we really need to know that? I mean for page content and post content, I don't really see folks using "main" groups unless they're doing it wrong and we should fallback to root sections (unless I'm missing something)

Reply:
That is 50% of what this PR tries to introduce the concept of sections in a container. Just as we have a grouping block name (core/group) for group actions we can have a section root. The other option is using settings.

> Do we really need to know that?

We need to make sure we don't need to know post_type in the block editor package, that is why we're using a way to find where the root of sections is indifferent of the user edititng a template or a post. So the final implementation has the option to set what the root of sections is and then the block editor acts accordingly. It does look more like a setting 🤷🏻  But yes we need to know.

Reply:
The aim of the feature is to allow users to add patterns to their pages at a high level. The trouble is that `page` means different things in different contexts. If I'm editing the home page template then I want to be able to add patterns to the `main` group. However if I am editing a page (not a the Pages template the actual page content), then I want to be able to add patterns to the post content block itself.

When we are in zoomed out mode then content inside the "container" aren't editable. If we just use `main` then when users are editing pages, the post content block will be treated as the top level block and the content inside it won't be editable, so I won't be able to build page content using this mode.

Reply:
> The aim of the feature is to allow users to add patterns to their pages at a high level. The trouble is that page means different things in different contexts. If I'm editing the home page template then I want to be able to add patterns to the main group. However if I am editing a page (not a the Pages template the actual page content), then I want to be able to add patterns to the post content block itself.

So this is about the "template-locked" mode right? because if I'm not wrong in the "post only" rendering mode, the post-content is top level so there's no need of a special case there.

Now if we do want to support zoom-out mode in "template-locked" mode (which is a bit weird to be honest), then yeah I agree we need something to define the "sections container" and in that case, I'd potentially add a new setting (private) to the `BlockEditorProvider` and eliminate both the logic to fetch "main" group from block editor package. The block editor package would always rely on that setting.

Reply:
Why do you think its weird to support zoom-out mode when we have a locked template? My understanding is that one of the primary uses for the zoom out view is to compose pages made of patterns. That seems more useful when editing a specific page than when editing templates IMO.

Reply:
> Why do you think its weird to support zoom-out mode when we have a locked template?

I don't know, I guess I expect the patterns to be mainly root level and in this case, there might be a lot of things around them. So in a sense, maybe zoom-out mode should trigger "post-only" rendering when you're actually editing pages or posts.

Reply:
> It is top level but without naming the block how do you know which top level block is the root you want to pick as the container for the sections

In my previous comment, when I said "top level" I meant "root level", so there's no need to pick a container, there's none.

Reply:
How so?

![image](https://github.com/WordPress/gutenberg/assets/107534/86bff085-2ad9-441b-bc8d-588e9c667aec)

Reply:
I was talking about "post-only" mode. Basically "disable template previewing" from the template sidebar panel.

Reply:
In that case, I'd argue for not trying to be "smart" in the block-editor package to "find" the right area. Instead I'd pass the "clientId" or the identifier of the area as a "setting" to `BlockEditorProvider`. Would that be possible?

Reply:
> Because zoom-out is a feature of the generic block editor package it has to work without knowing the implementation details of WordPress

Just throwing some ideas out, but it could be that the block editor's zoomed out mode provides some base functionality (the canvas zooming, potentially just as an API), and the editor package has some additional enhancements on top of that (the button that triggers zoomed out mode, figuring out the root block, block editor mode locking).

Reply:
> figuring out the root block
This is how this PR does it.

> to be "smart" in the block-editor package to "find" the right area
This PR avoids that by allowing final implementations to set the right area.